src/main/java/qupath/ext/align/gui/ImageAlignmentPane.java

```diff
@@ -517,38 +523,97 @@ void promptToAddImages() {
 				images.stream().map(i -> project.getEntry(i)).collect(Collectors.toSet());
 		if (currentEntry != null)
 			alreadySelected.remove(currentEntry);
-		
+
+		entries.removeAll(alreadySelected);
+				
 		// Create a list to display, with the appropriate selections
-		ListView<ProjectImageEntry<BufferedImage>>  list = new ListView<>();
-		list.getItems().setAll(entries);
-		list.getSelectionModel().setSelectionMode(SelectionMode.MULTIPLE);
+		ListSelectionView<ProjectImageEntry<BufferedImage>>  list = new ListSelectionView<>();
+		list.getSourceItems().setAll(entries);
+
+		//list.getSelectionModel().setSelectionMode(SelectionMode.MULTIPLE);
+		/*
 		for (int i = 0; i < entries.size(); i++) {
 			if (alreadySelected.contains(entries.get(i)))
 				list.getSelectionModel().select(i);
 		}
+		*/
+		list.setCellFactory(c -> new ProjectEntryListCell());
+
+		// Add a filter text field
+		TextField tfFilter = new TextField();
+		tfFilter.textProperty().bindBidirectional(filterText);
+		filterText.addListener((v, o, n) -> updateImageList(list, entries, alreadySelected, n));
+
+		if (tfFilter.getText() != "")
```

Comment:
https://www.baeldung.com/java-compare-strings

Reply:
I'm not sure if it's also possible to have `null`, or blank (but non-empty) strings.
(FWIW I recently learned about [`getValueSafe()`](https://openjfx.io/javadoc/18/javafx.base/javafx/beans/binding/StringExpression.html#getValueSafe()) for `StringProperty`, which helps a bit with the first of these)